### PR TITLE
Merged files, removed pmap_experimental.

### PR DESCRIPTION
**Pull request by AlexanderMath**
Removing `pmap_experimental_nanoDFT` by merging changes into `nanoDFT`. All the jax.pmap stuff becomes identity when ndevices=1.

### CODE REVIEW

File: pyscf_ipu/nanoDFT/nanoDFT.py [hunk 1]
```diff
@@ -59,60 +72,78 @@ def nanoDFT_iteration(i, vals, opts, mol):
...
+
+    return diff_JK
 
-def _nanoDFT(state, opts, mol):
+def _nanoDFT(state, ERI, grid_AO, grid_weights, opts, mol):
     # Utilize the IPUs MIMD parallism to compute the electron repulsion integrals (ERIs) in parallel.
-    if opts.backend == "ipu": state.ERI = electron_repulsion_integrals(state.input_floats, state.input_ints, mol, opts.threads_int, opts.intv)
-    else: pass # Compute on CPU.
+    #if opts.backend == "ipu": state.ERI = electron_repulsion_integrals(state.input_floats, state.input_ints, mol, opts.threads_int, opts.intv)
+    #else: pass # Compute on CPU.
```
Comment (awf) [inline pyscf_ipu/nanoDFT/nanoDFT.py:126]: Maybe lose the comment at line 124 too? I.e. delete all of this.

File: pyscf_ipu/nanoDFT/nanoDFT.py [hunk 2]
```diff
@@ -473,18 +561,16 @@ def nanoDFT_options(
         threads_int (int): For -backend ipu. Number of threads for computing ERI with custom C++ (trades off speed vs memory).
         eri_threshold (float): Zero out ERIs that are below the threshold in absolute value. Not supported for '--backend ipu'. 
         ao_threshold (float): Zero out grid_AO that are below the threshold in absolute value.
+        dense_ERI (bool): Whether to use dense ERI (s1) or sparse symmtric ERI.
```
Comment (awf) [inline pyscf_ipu/nanoDFT/nanoDFT.py:564]:
```suggestion
        dense_ERI (bool): Whether to use dense ERI (s1) or sparse symmtric ERI.
        batches (int): Number of ERIs per batch per IPU
        ndevices (int) : Number of IPUs to use
        v (bool): Verbose 
        profile (bool): If we only want profile exit after IPU finishes. 
```

**Review by awf — APPROVED** [review]
Looks good - great to have these unified.  I've read in reasonable detail but have only some doc changes to propose.

File: pyscf_ipu/nanoDFT/nanoDFT.py (hunk 1 above)
Comment (AlexanderMath) [reply pyscf_ipu/nanoDFT/nanoDFT.py:126]: @mihaipgc  is working on rewriting `electron_repulsion_integrals(..)` to fit the layout in `sparse_symmetric_ERI(..)`; when we get that we'll revert back to these lines. 

After this, when @balancap allow tesselate to shard `jax.lax.gather_p` and `jax.lax.scater_add_p`, the DFT will never move ERI indices and ERI floats between tiles.

**Comment by hatemhelal** [comment]
> @hatemhelal IPU_MODEL fails when using int16/float16 for the indices. Can we do something like `if ipu_model: dense_ERI=True`?

yes I think the way to do it would be:

 ```
pyscf_ipu.experimental.device.has_ipu() and "JAX_IPU_USE_MODEL" in os.environ
```

It is handy to have utilities for feature testing if we are running on an IPU and if that IPU is a model so seems worthwhile to put that in a  common place.

**Pull request merged**